PROTO_tests/tests/segarray_test.py

```diff
@@ -0,0 +1,694 @@
+        exp_a = np.concatenate([np.array([a[0]]), np.array(a[1:])[~exp_idx_a]])
+        exp_b = np.concatenate([np.array([b[0]]), np.array(b[1:])[~exp_idx_b]])
+
+        # Test with no empty segments
+        segments = ak.array([0, len(a)])
+        flat = ak.array(a + b)
+        sa = ak.SegArray(segments, flat)
+        result = sa.remove_repeats()
+        assert [0, len(exp_a)] == result.segments.to_list()
+        assert np.concatenate([exp_a, exp_b]).tolist() == result.values.to_list()
+
+        # test empty segments
+        # TODO - update line below to segments = ak.array([0, 0, len(a), len(a), len(a), len(a)+len(b)])
+        #  when issue #2661 is corrected. Also, needed in the first assert below
+        segments = ak.array([0, len(a), len(a), len(a), len(a) + len(b)])
+        flat = ak.array(a + b)
+        sa = ak.SegArray(segments, flat)
+        result = sa.remove_repeats()
+        assert [
+            0,
+            len(exp_a),
+            len(exp_a),
+            len(exp_a),
+            len(exp_a) + len(exp_b),
+        ] == result.segments.to_list()
+        assert np.concatenate([exp_a, exp_b]).tolist() == result.values.to_list()
+
+    @pytest.mark.parametrize("dtype", NO_FLOAT_STR)
+    @pytest.mark.parametrize("op", SETOPS)
+    def test_setops(self, dtype, op):
+        # TODO - string results not properly ordered. Need to add ak.str_ testing
+        #  back once #2665 is worked.
+        a, b, c, d = self.get_setops_segments(dtype)
+
+        # test with no empty segments
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr_2 = ak.SegArray(ak.array([0, len(c)]), ak.array(c + d))
+        sa_op = getattr(segarr, op)
+        result = sa_op(segarr_2)
+
+        np_func = getattr(np, f"{op}1d")
+        exp_1 = np_func(np.array(a), np.array(c))
+        exp_2 = np_func(np.array(b), np.array(d))
+        assert result.segments.to_list() == [0, len(exp_1)]
+        assert result.values.to_list() == np.concatenate([exp_1, exp_2]).tolist()
+        exp_sa = ak.SegArray(ak.array([0, len(exp_1)]), ak.array(np.concatenate([exp_1, exp_2])))
+        assert result.to_list() == exp_sa.to_list()
+
+        # TODO - empty segments testing
+
+    @staticmethod
+    def get_filter(dtype):
+        if dtype in [ak.int64, ak.uint64]:
+            return 3
+        elif dtype == ak.bigint:
+            return 3 + 2**200
+        elif dtype == ak.float64:
+            return 3.14
+        elif dtype == ak.bool:
+            return False
+        elif dtype == ak.str_:
+            return "a"
+
+    @staticmethod
+    def get_filter_list(dtype):
+        if dtype in [ak.int64, ak.uint64]:
+            return [1, 3]
+        elif dtype == ak.bigint:
+            return [1 + 2**200, 3 + 2**200]
+        elif dtype == ak.float64:
+            return [3.14, 1.01]
+        elif dtype == ak.bool:
+            return [False]
+        elif dtype == ak.str_:
+            return ["a", "abc"]
```

Comment:
i just notice `ak.bool` is the only one with a single element, should we make it `return [False, True]`? Or is this intentional because that would filter out all the values?

Reply:
This is intentional. It being a list is the only thing that should matter here.